tooling/composer-cache

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+set -e
+
+COMPOSER_WARMUP=~/.dd-trace-php-composer-warmup
+
+function usage {
+    echo "Usage:"
+    echo "  ./composer-update-all.sh cache"
+    echo "  ./composer-update-all.sh link-cache"
+    echo "  ./composer-update-all.sh unlink-cache"
+    echo
+}
+
+# If command (arg 1) is not provided, let's print the usage message and exit with an error code.
+if [[ ! $1 ]]
+then
+    usage
+    exit 1
+fi
+
+# Cache all the composer.lock files and vendor folders for future quick reuse
+cache() {
+    echo "Ensuring clean warmup dir '${COMPOSER_WARMUP}' exists (it might take a while)."
+    rm -rf ${COMPOSER_WARMUP}
+    mkdir -p ${COMPOSER_WARMUP}
+
+    for composer_json in $(find tests/Frameworks/ -name composer.json | grep -v "/vendor" | grep -v "CakePHP")
```

Comment:
Why do we ignore CakePHP?

Reply:
It has some strange behavior that I need to investigate further and I need to de-prioritize at the moment. I created a separate task for this if you are fine with that. So we can merge this PR and be much faster with build time.

I do not want to push this too much, so if you believe that I should better look into this let's discuss it.